### ISSUE TITLE
Weighted Graph support?

### ISSUE THREAD

**Issue by ReichYang** [mbojan/netseg#17]
Hi, first of all, thank you for writing this package.

I'm interested in the homophily in my directed and weighted graph with certain node attributes. In my impression, all the existing implementations of the metrics are for unweighted graph, right? Is there a way to calculate the metrics taking into the weights of the edges?

**Comment by mbojan** [comment]
Thanks @ReichYang , Yes and no, so to speak. It depends on:

1. What are the "weights" in your data? E.g. are they counts?
2. Which measure you'd be after?

EDIT (expanding). Most of the indices are functions of the mixing matrix. In fact, they are also implemented as such, that is have methods for the data to be supplied not as a network/igraph object but as matrix or array. You can compute the mixing matrix taking the edge weights into account in more or less arbitrary manner (that's why I ask what are they above) and compute one of the mixing-matrix-based measures. Not all of the measures are based on mixing matrix though (that's why I asked which measure you are after above).

**Comment by ReichYang** [comment]
> Thanks @ReichYang , Yes and no, so to speak. It depends on:
> 
> 1. What are the "weights" in your data? E.g. are they counts?
> 2. Which measure you'd be after?
> 
> EDIT (expanding). Most of the indices are functions of the mixing matrix. In fact, they are also implemented as such, that is have methods for the data to be supplied not as a network/igraph object but as matrix or array. You can compute the mixing matrix taking the edge weights into account in more or less arbitrary manner (that's why I ask what are they above) and compute one of the mixing-matrix-based measures. Not all of the measures are based on mixing matrix though (that's why I asked which measure you are after above).

Hi, thanks for the reply. 
1. I have edge weights, for example, a link between A and B is counted as 1 in an unweighted network, but in my case, I have values that are more than binary.
2. I prefer to see all homophily measures supporting that. 

I did try one approach where I feed in each function with a weighted mixing matrix I calculated on my own--but I'm not sure what should be the gsizes, should they still be the same as an unweighted network, like the number of categories for my node attribute of interest?

**Comment by mbojan** [comment]
> 1. I have edge weights, for example, a link between A and B is counted as 1 in an unweighted network, but in my case, I have values that are more than binary.

That's what ppl usually mean by weights -- values on the edges. The question is what do they represent? If they are or behave like counts then the approach with mixing matrix I described earlier will make sense. It might not make sense in other contexts.


> 2. I prefer to see all homophily measures supporting that.

I'm sure you do. This is not a technical/implementation problem but conceptual. I quietly recommended skimming through https://doi.org/10.1016/j.socnet.2014.04.001 and familiarize with how these measures work. The "weights" in networks of valued ties can have very different nature and homophily/segregation measurement will require different approaches -- not unlike different scales of measurement require different association measures (Pearson, Kendall, Spearman etc.). 



> I did try one approach where I feed in each function with a weighted mixing matrix I calculated on my own--but I'm not sure what should be the gsizes, should they still be the same as an unweighted network, like the number of categories for my node attribute of interest?

That's what I suggested. Do mind that it is not feasible for all the measures (as I write above).

The `gsizes` argument expects the vector of group sizes (as documented) -- numbers of nodes in each group. Edge weights do not influence these.
